server/neptune/terraform/async_client.go

```diff
@@ -0,0 +1,194 @@
+package terraform
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"os/exec"
+	"regexp"
+	"sync"
+
+	"github.com/hashicorp/go-version"
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/core/runtime/cache"
+	"github.com/runatlantis/atlantis/server/core/terraform"
+	"github.com/runatlantis/atlantis/server/core/terraform/helpers"
+	"github.com/runatlantis/atlantis/server/logging"
+)
+
+// Setting the buffer size to 10mb
+const BufioScannerBufferSize = 10 * 1024 * 1024
+
+// versionRegex extracts the version from `terraform version` output.
+//     Terraform v0.12.0-alpha4 (2c36829d3265661d8edbd5014de8090ea7e2a076)
+//	   => 0.12.0-alpha4
+//
+//     Terraform v0.11.10
+//	   => 0.11.10
+var versionRegex = regexp.MustCompile("Terraform v(.*?)(\\s.*)?\n")
+
+func NewAsyncClient(
+	binDir string,
+	cacheDir string,
+	defaultVersionStr string,
+	defaultVersionFlagName string,
+	tfDownloadURL string,
+	tfDownloader terraform.Downloader,
+	usePluginCache bool,
+) (*AsyncClient, error) {
+	version, err := getDefaultVersion(defaultVersionStr, defaultVersionFlagName)
+	if err != nil {
+		return nil, errors.Wrapf(err, "getting default version")
+	}
+
+	loader := terraform.NewVersionLoader(&terraform.DefaultDownloader{}, tfDownloadURL)
+
```

Comment:
the `github.com/runatlantis/atlantis/server/core/terraform` pkg has a lot of things going on that I think we should eventually break down. Can you instead pull out what you need into neptune like the async client?

Reply:
I just reused the version loader as it is since we didn't have to tweak the implementation, unlike async client which was tightly coupled with our legacy code (i.e `projectContext` and `projectCommandOutputHandler`). I don't see value in duplicating the code for Neptune if it is going to be exactly the same. 

However, it makes sense to do so If we plan to keep them isolated entirely. @nishkrishnan what do you think?

Reply:
yeah fine to just move that stuff later